### PR TITLE
[cmake] add targets for cppcheck and clang-tidy

### PR DESCRIPTION
**Pull request by lrusak**
This adds targets for running cppcheck and clang-tidy specifically for output used for jenkins.

Example reports can be found here:
https://jenkins.kodi.tv/view/Reports/job/LINUX-64-Static-Analysis/1/cppcheck/
https://jenkins.kodi.tv/view/Reports/job/LINUX-64-Static-Analysis/1/clang-tidy/

These reports could be done for the apple projects as well.

These reports are useful to have as they are public (coverity reports are private).

### CODE REVIEW

**Review by phunkyfish — APPROVED** [review]
Nice, I love this addition

**Review by Rechi — COMMENTED** [review]
The correct way is to set [CMAKE_\<LANG>_CLANG_TIDY](https://cmake.org/cmake/help/v3.6/variable/CMAKE_LANG_CLANG_TIDY.html) and [CMAKE_\<LANG>_CPPCHECK](https://cmake.org/cmake/help/v3.10/variable/CMAKE_LANG_CPPCHECK.html) to get the info like compiler warnings in the build log.
The additional targets `analyze-clang-tidy` and `analyze-cppcheck` can be kept for CI.

**Comment by wsnipex** [comment]
we already have cppcheck support: https://github.com/xbmc/xbmc/pull/12543

**Comment by lrusak** [comment]
> The correct way is to set [CMAKE_<LANG>_CLANG_TIDY](https://cmake.org/cmake/help/v3.6/variable/CMAKE_LANG_CLANG_TIDY.html) and [CMAKE_<LANG>_CPPCHECK](https://cmake.org/cmake/help/v3.10/variable/CMAKE_LANG_CPPCHECK.html) to get the info like compiler warnings in the build log.
> The additional targets `analyze-clang-tidy` and `analyze-cppcheck` can be kept for CI.

I only want it to run if calling those targets.



> we already have cppcheck support: #12543

I'm aware :+1:

**Comment by AlwinEsch** [comment]
Looks good to have it in a special cmake file for.
Can be on this file also added in future a "C" check?

Thought about the "C" part on binary addon headers, so that on every Kodi compile, it becomes checked and confirmed that no "C++" parts move there.

Currently are still some C++ parts inside them, where not allow check for now. But after I'm all of them have fixed, can then a check added to there from me :smile:.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
seems to be a bug in jenkins that after a blacklisted label was added (and removed) jenkins won't build from the webhook.

Successful build is here: https://jenkins.kodi.tv/job/BuildMulti-PR-Manually/465/

Please review so we can merge 😄

File: cmake/scripts/common/StaticAnalysis.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
+include(ProcessorCount)
+ProcessorCount(CPU_CORES)
+
+find_program(CPPCHECK cppcheck)
```
Comment (Rechi) [inline cmake/scripts/common/StaticAnalysis.cmake:4]:
```suggestion
find_program(CPPCHECK_EXECUTABLE cppcheck)
```

File: cmake/scripts/common/StaticAnalysis.cmake [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+if(CPPCHECK)
+  add_custom_target(analyze-cppcheck
+    DEPENDS ${APP_NAME_LC} ${APP_NAME_LC}-test
+    COMMAND ${CPPCHECK}
```
Comment (Rechi) [inline cmake/scripts/common/StaticAnalysis.cmake:9]:
```suggestion
    COMMAND ${CPPCHECK_EXECUTABLE}
```

File: cmake/scripts/common/StaticAnalysis.cmake [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            --xml
+            --xml-version=2
+            --language=c++
+            --output-file=${CMAKE_BINARY_DIR}/cppcheck-result.xml
+            xbmc
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+    COMMENT "Static code analysis using cppcheck")
+endif()
+
+find_program(CLANG_TIDY NAMES clang-tidy)
```
Comment (Rechi) [inline cmake/scripts/common/StaticAnalysis.cmake:26]:
```suggestion
find_program(CLANG_TIDY_EXECUTABLE clang-tidy)
```

File: cmake/scripts/common/StaticAnalysis.cmake [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+find_program(RUN_CLANG_TIDY NAMES run-clang-tidy.py run-clang-tidy
+                            PATHS /usr/share/clang /usr/bin)
+
+if(RUN_CLANG_TIDY AND CLANG_TIDY)
```
Comment (Rechi) [inline cmake/scripts/common/StaticAnalysis.cmake:30]:
```suggestion
if(RUN_CLANG_TIDY AND CLANG_TIDY_EXECUTABLE)
```

File: cmake/scripts/common/StaticAnalysis.cmake [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  add_custom_target(analyze-clang-tidy
+    DEPENDS ${APP_NAME_LC} ${APP_NAME_LC}-test
+    COMMAND ${RUN_CLANG_TIDY}
+            -j${CPU_CORES}
+            -clang-tidy-binary=${CLANG_TIDY}
```
Comment (Rechi) [inline cmake/scripts/common/StaticAnalysis.cmake:35]:
```suggestion
            -clang-tidy-binary=${CLANG_TIDY_EXECUTABLE}
```

**Review by Rechi — COMMENTED** [review]
Please use the same variable names for cppcheck and clang-tidy as in the find modules.

**Comment by lrusak** [comment]
jenkins build this please

**Pull request merged**